PacklinkPro/Setup/UpgradeSchema.php

```diff
@@ -119,4 +143,105 @@ protected function changeShipmentDataUpdateInterval()
 
         Logger::logInfo('Update script V1.0.1 has been successfully completed.');
     }
+
+    /**
+     * Enqueues task for cleaning up completed queue items.
+     *
+     * @throws \Packlink\PacklinkPro\IntegrationCore\Infrastructure\ORM\Exceptions\RepositoryNotRegisteredException
+     */
+    protected function addTaskCleanupSchedule()
+    {
+        Logger::logInfo('Started executing V1.1.0 update script.');
+
+        $configuration = ServiceRegister::getService(Configuration::CLASS_NAME);
+        try {
+            $repository = RepositoryRegistry::getRepository(Schedule::getClassName());
+        } catch (RepositoryNotRegisteredException $e) {
+            Logger::logError("V1.1.0 update script failed because: {$e->getMessage()}");
+
+            throw $e;
+        }
+
+        $schedule = new HourlySchedule(
+            new TaskCleanupTask(ScheduleCheckTask::getClassName(), [QueueItem::COMPLETED], 3600),
+            $configuration->getDefaultQueueName()
+        );
+
+        $schedule->setMinute(10);
+        $schedule->setNextSchedule();
+        $repository->save($schedule);
+
+        Logger::logInfo('Update script V1.1.0 has been successfully completed.');
```

Comment:
It is more readable this way, in my opinion.

Reply:
I think implement the normal flow into try statement and error flow in catch statement could be better . When we  implement the correct flow in two separate statements we are breaking in several parts the same logic ,

Reply:
Again, the try/catch block only tries to detect a specific type of exception. If it passes, the storage is available and we can continue. When one looks at the code, there is a separate block that handles this. This block could be easily observed and maybe even extracted to a separate block (a method, function, etc.). The exception that could occur in the rest of the code is not intended to be handled with the catch statement above, but with a more global exception handler (outside the method). If we would have wrapped all in try/catch, it would indicate that we want to catch the specific exception in all the code, which we don't, since it can only occur in the wrapped block. This is why we use try/catch block only for code that can actually throw the caught exception.

Reply:
If you only use exception as check if this function can be executed then it probably need to be implement outside the function , the caller should have responsibility to choose when this function (without check) will be executed . In other hand if you want to implement this check into this function and throw exception the better way is with one try-catch .

Reply:
Restructured the code to handle more generic exceptions during the upgrade process.